### PR TITLE
Impove the system image build script

### LINKED ISSUE #1673: Ignore dev-packages when building sysimage
When building a custom sysimage, we should ignore packages in "dev" mode, since they they are likely to change without affecting the manifest.

I'm not sure what the canonical way to identify dev-packages is - in my own scripts I'm currently using `Pkg.Operations.manifest_info.(ctx, uuid).path != nothing`, which seems to do the job.

### PR DESCRIPTION
**Pull request by zhubonan**
Fixes #1673

For every PR, please check the following:
- [x] End-user documentation check. If this PR requires end-user documentation, please add that at https://github.com/julia-vscode/docs.
- [x] Changelog mention. If this PR should be mentioned in the CHANGELOG, please edit the CHANGELOG.md file in this PR.

Default to exclude development packages during system image building. Also, allow passing additional options to the `PackageCompiler.create_sysimge` function based on a `vscode-compileenv.toml` file located inside the project root folder. 
This can be used to pass precompile execution/statement files and further reduced the first-call latency.

### CODE REVIEW

File: scripts/tasks/task_compileenv.jl [hunk 1]
```diff
@@ -1,4 +1,59 @@
...
+        "path" in keys(sub[1]) && push!(devpkgs, Symbol(key))
+    end
+    devpkgs
+end
+
+
+"""
+    read_configuration(envdir)
+
+Read the configuration file `vscode-compileenv.toml` from the environment directory.
```
Comment (pfitzseb) [inline scripts/tasks/task_compileenv.jl:28]: Considering this is VS Code specific, should we maybe use `./.vscode/JuliaSysimg.toml` instead?

File: scripts/tasks/task_compileenv.jl [hunk 2]
```diff
@@ -1,4 +1,59 @@
...
+
+This file should look like:
+
+```
+[sysimage]
+excluded_packages=[]   # Additional packages to be exlucded in the system image
+precompile_statements_file=[]  # Precompile statements file to be used
+precompile_execution_file=[] # Precompile execution file to be used
```
Comment (pfitzseb) [inline scripts/tasks/task_compileenv.jl:36]:
```suggestion
exclude = []   # Additional packages to be exlucded in the system image
statements_files = []  # Precompile statements file to be used
execution_files = [] # Precompile execution file to be used
```
maybe?

File: scripts/tasks/task_compileenv.jl [hunk 3]
```diff
@@ -8,8 +63,18 @@ project_filename = isfile(joinpath(env_to_precompile, "JuliaProject.toml")) ? jo
...
+config = read_configuration(env_to_precompile)
+dev_packages = find_dev_packages(env_to_precompile)
+
+# Assemble the arguments for the `create_sysimage` function
+used_packages = filter(x -> !(x in dev_packages || x in config[:excluded_packages]), Symbol.(collect(keys(project.deps))))
+precompile_statements = config[:precompile_statements_file]
+precompile_execution = config[:precompile_execution_file]
 
-@info "Now building a custom sysimage for the environment '$env_to_precompile'."
+@info "Now building a custom sysimage for the environment '$env_to_precompile', excluding dev packages '$dev_packages'."
```
Comment (pfitzseb) [inline scripts/tasks/task_compileenv.jl:75]: Might be better to show a list of packages that get compiled instead?

**Review by pfitzseb — COMMENTED** [review]
Very cool! Got a few comments, but nothing critical.

File: scripts/tasks/task_compileenv.jl (hunk 1 above)
Comment (zhubonan) [reply scripts/tasks/task_compileenv.jl]: Thanks. I have updated the file path to `./.vscode/JuliaSysimg.toml`.

File: scripts/tasks/task_compileenv.jl (hunk 2 above)
Comment (zhubonan) [reply scripts/tasks/task_compileenv.jl]: good idea! I have applied the changes in the code.

File: scripts/tasks/task_compileenv.jl (hunk 3 above)
Comment (zhubonan) [reply scripts/tasks/task_compileenv.jl:75]: Indeed, the script now prints out a list of included packages.

**Comment by zhubonan** [comment]
@pfitzseb this PR is ready for review now! Thanks

**Comment by oschulz** [comment]
Thanks for this, @zhubonan !

**Pull request merged**